### DISCUSSION TITLE
rust-course/Once 的使用

### DISCUSSION THREAD

**Discussion by nami-bu** [sunface/rust-course#455] [内容建议]
我本地使用的是 Rust 1.57.0，看到 《Rust圣经》中多线程并发-使用多线程这个章节的 "[只被调用一次的函数]"时，发现例子跑不通，根据错误提示改成了如下代码：
use std::sync::Once;
use std::thread;

static mut VAL: usize = 0;
static INIT: Once= Once::new();

fn main() {
    let handle1 = thread::spawn(move || {
        INIT.call_once(|| {
            unsafe {
                VAL = 1;
            }
        });
    });

    let handle2 = thread::spawn(move || {
        INIT.call_once(|| {
            unsafe {
                VAL = 2;
            }
        });
    });

    handle1.join().unwrap();
    handle2.join().unwrap();

    println!("{}", unsafe { VAL });
}
就可以了。这个看下要不要后续补充到文档中

**Comment by sunface** [comment]
非常棒，宝贵的建议值得以一个 pr 来收尾，欢迎提交 :)
